samples/encode-decode-worker/js/stream_worker.js

```diff
@@ -0,0 +1,323 @@
+               this.decoder.decode(chunk);
+               const after = performance.now();
+               const duration = after - before;
+               dec_update(duration);
+             } catch (e) {
+               self.postMessage({severity: 'fatal', text: 'Derror size: ' + chunk.byteLength + ' seq: ' + chunk.seqNo + ' kf: ' + chunk.keyframeIndex + ' delta: ' + chunk.deltaframeIndex + ' dur: ' + chunk.duration + ' ts: ' + chunk.timestamp + ' ssrc: ' + chunk.ssrc + ' pt: ' + chunk.pt + ' tid: ' + chunk.temporalLayerId + ' type: ' + chunk.type});
+               self.postMessage({severity: 'fatal', text: `Catch Decode error: ${e.message}`});
+             }
+           }
+         }
+       }
+     });
+   }
+
+   EncodeVideoStream(self, config) {
+     return new TransformStream({
+       start(controller) {
+         this.frameCounter = 0;
+         this.seqNo = 0;
+         this.keyframeIndex = 0;
+         this.deltaframeIndex = 0;
+         this.pending_outputs = 0;
+         this.encoder = encoder = new VideoEncoder({
+           output: (chunk, cfg) => {
+             if (cfg.decoderConfig) {
+               // self.postMessage({text: 'Decoder reconfig!'});
+               const decoderConfig = JSON.stringify(cfg.decoderConfig);
+               // self.postMessage({text: 'Configuration: ' + decoderConfig});
+               const configChunk =
+               {
+                  type: "config",
+                  seqNo: this.seqNo,
+                  keyframeIndex: this.keyframeIndex,
+                  deltaframeIndex: this.deltaframeIndex,
+                  timestamp: 0,
+                  pt: 0,
+                  config: decoderConfig 
+               };
+               controller.enqueue(configChunk); 
+             } 
+             chunk.temporalLayerId = 0;
+             if (cfg.svc.temporalLayerId) {
+               chunk.temporalLayerId = cfg.svc.temporalLayerId;
+             }
+             this.seqNo++;
+             if (chunk.type == 'key') {
+               this.keyframeIndex++;
+               this.deltaframeIndex = 0;
+             } else {
+               this.deltaframeIndex++;
+             } 
+             this.pending_outputs--;
+             chunk.seqNo = this.seqNo;
+             chunk.keyframeIndex = this.keyframeIndex;
+             chunk.deltaframeIndex = this.deltaframeIndex;
+             controller.enqueue(chunk);
+           },
+           error: (e) => {
+             self.postMessage({severity: 'fatal', text: `Encoder error: ${e.message}`});
+           }
+         });
+         VideoEncoder.isConfigSupported(config).then((encoderSupport) => {
+           if(encoderSupport.supported) {
+             this.encoder.configure(encoderSupport.config);
+             self.postMessage({text: 'Encoder successfully configured:\n' + JSON.stringify(encoderSupport.config)});
+             // self.postMessage({text: 'Encoder state: ' + JSON.stringify(this.encoder.state)});
```

Comment:
Ditto.

Reply:
This message isn't particularly useful even for debugging, so I've taken it out.